File: budgetz-web-api/Models/Expense.cs
namespace Budgetz.Models;
public enum RecurringType
{
    None = 0,
    Monthly = 1,
    Yearly = 2
}
public enum ExpenseType
{
    Expense = 0,
    Income = 1
}

public class Expense : EntityBase
{
    public string Name { get; set; }

    public double Amount { get; set; }

    public DateTimeOffset Date { get; set; }
    public string Month { get; set; }
    public string Category { get; set; }
    public RecurringType Recurring { get; set; }
    public ExpenseType Type { get; set; }

}

File: budgetz-app/src/app/models/Expense.ts
export enum RecurringType {
  None = 0,
  Monthly = 1,
  Yearly = 2
}

export enum ExpenseType {
  Expense = 0,
  Income = 1
}

export interface Expense {
  id?: number;
  name: string;
  amount: number;
  month: string;
  category: string;
  recurring?: RecurringType;
  type: ExpenseType;
}